### PR TITLE
[WIP] Merge Fetch15 branch

### PR DESCRIPTION
**Pull request by knorth55**
Move from #1152 
Current fetch15 branch.
This branch have these features follows;

## Go to kitchen
- [ ] #1149 

## Others

- [ ] #1225
- [ ] #1236
- [ ] #1312
- [ ] #1323
- [ ] #1332
- [ ] #1337
- [ ] #1342
- [ ] #1422
- [ ] #1460
- [ ] #1463 
- [ ] #1465
- [ ] #1466
- [ ] #1480
- [ ] #1492
- [ ] #1527 
- [x] #1561 
- [x] #1568 
- [x] #1571 

## Upstream depencency
- `jsk-ros-pkg/jsk_pr2eus`
  - [ ] https://github.com/jsk-ros-pkg/jsk_pr2eus/pull/406
- `jsk-ros-pkg/jsk_3rdparty`
  - [ ] https://github.com/jsk-ros-pkg/jsk_3rdparty/pull/337

## New package dependency
- https://github.com/knorth55/app_manager_utils
- https://github.com/knorth55/coral_usb_ros

## Merged
- `jsk-ros-pkg/jsk_robot`
  - [x] #1128
  - [x] #1147 
  - [x] #1148 
  - [x] #1150
  - [x] #1151 
  - [x] #1159
  - [x] #1160
  - [x] #1166 
  - [x] #1167
  - [x] #1168 
  - [x] #1169 
  - [x] #1174
  - [x] #1188
  - [x] #1193 
  - [x] #1194 
  - [x] #1195
  - [x] #1199 
  - [x] #1200
  - [x] #1201
  - [x] #1202
  - [x] #1204
  - [x] #1205 
  - [x] #1206
  - [x] #1208 
  - [x] #1209 
  - [x] #1219
  - [x] #1220
  - [x] #1224 
  - [x] #1226 
  - [x] #1231
  - [x] #1232 
  - [x] #1234 
  - [x] #1237
  - [x] #1239
  - [x] #1240
  - [x] #1250
  - [x] #1252
  - [x] #1253
  - [x] #1254
  - [x] #1255
  - [x] #1256
  - [x] #1257
  - [x] #1260
  - [x] #1261
  - [x] #1267
  - [x] #1268
  - [x] #1269 
  - [x] #1276
  - [x] #1277 
  - [x] #1278
  - [x] #1280
  - [x] #1285 
  - [x] #1291
  - [x] #1292
  - [x] #1293
  - [x] #1295
  - [x] #1296
  - [x] #1309
  - [x] #1310
  - [x] #1311
  - [x] #1315
  - [x] #1316 
  - [x] #1317
  - [x] #1321
  - [x] #1327
  - [x] #1330
  - [x] #1338
  - [x] #1352
  - [x] #1359
  - [x] #1363
  - [x] #1366
  - [x] #1367
  - [x] #1368
  - [x] #1369
  - [x] #1370
  - [x] #1371
  - [x] #1374
  - [x] #1383
  - [x] #1384
  - [x] #1385
  - [x] #1386 
  - [x] #1394
  - [x] #1396
  - [x] #1400
  - [x] #1403
  - [x] #1405
  - [x] #1416
  - [x] #1431
  - [x] #1435
  - [x] #1449
  - [x] #1450
  - [x] #1451
  - [x] #1452
  - [x] #1453
  - [x] #1454
  - [x] #1455
  - [x] #1456
  - [x] #1457
  - [x] #1458
  - [x] #1459
  - [x] #1461
  - [x] #1462
  - [x] #1464
  - [x] #1467
  - [x] #1481
  - [x] #1485 
  - [x] #1487
  - [x] #1500
  - [x] #1508 
  - [x] #1519 
  - [x] #1532 
  - [x] #1533
  - [x] #1534 

- `jsk-ros-pkg/jsk_pr2eus`
  - [x] https://github.com/jsk-ros-pkg/jsk_pr2eus/pull/418
- `jsk-ros-pkg/jsk_3rdparty`
  - [x] https://github.com/jsk-ros-pkg/jsk_3rdparty/pull/221
- `jsk-ros-pkg/jsk_demos`
  - [x] https://github.com/jsk-ros-pkg/jsk_demos/pull/1319
  - [x] https://github.com/jsk-ros-pkg/jsk_demos/pull/1329
- `fetchrobotics/fetch_ros`
  - [x] https://github.com/fetchrobotics/fetch_ros/pull/137
  - [x] https://github.com/fetchrobotics/fetch_ros/pull/144
  - [x] https://github.com/fetchrobotics/fetch_ros/pull/146
- `PR2/app_manager`
  - [x] https://github.com/PR2/app_manager/pull/25
  - [x] https://github.com/PR2/app_manager/pull/49
- `ros-planning/navigation`
  - [x] https://github.com/ros-planning/navigation/pull/1039
- `tork-a/roswww`
  - [x] https://github.com/tork-a/roswww/pull/51
- `jsk-ros-pkg/jsk_common`
  - [x] https://github.com/jsk-ros-pkg/jsk_common/pull/1672
  - [x] https://github.com/jsk-ros-pkg/jsk_common/pull/1681

cc. @708yamaguchi @sktometometo @mqcmd196 @tkmtnt7000

### CODE REVIEW

**Comment by knorth55** [comment]
Fetch15のMelodic対応をマージしました．
https://github.com/knorth55/jsk_robot/pull/60

File: jsk_fetch_robot/jsk_fetch_startup/scripts/updata_workspace.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+. $HOME/ros/melodic/devel/setup.bash
+
+set -x
+
+cd $HOME/ros/melodic/src
+ln -sf $(rospack find jsk_fetch_startup)/../jsk_fetch.rosinstall.$ROS_DISTRO $HOME/ros/melodic/src/.rosinstall
+wstool foreach --git 'git stash'
+wstool update --delete-changed-uris
+catkin build
```
Comment (k-okada) [inline jsk_fetch_robot/jsk_fetch_startup/scripts/updata_workspace.sh:11]: catkin config かinit のときに，` -DCMAKE_BUILD_TYPE=Release` しているかな？
あるいは，
```
catkin build --cmake-args -DCMAKE_BUILD_TYPE=Release
```
にするのかな．

触っていて思うのはかなり体内PCが重くて色々なループが回っていない印象があるんだけど，そんなことないかな．．

気休めかもしれないけど，eigen 多用しているコードだと，かなり影響は大きいはずです．

Comment (k-okada) [reply jsk_fetch_robot/jsk_fetch_startup/scripts/updata_workspace.sh:11]: fyi: https://github.com/tu-darmstadt-ros-pkg/hector_slam/pull/32#issuecomment-801808247

**Comment by knorth55** [comment]
I rebase this PR to origin/master.
@708yamaguchi can you check this PR and split into several PRs if you find forgetting commits?

**Comment by knorth55** [comment]
@708yamaguchi @sktometometo @mqcmd196 can you help me to split this one PR to several PRs?
some commits are not submitted as PRs.
we cannot miss this chance!

**Comment by 708yamaguchi** [comment]
thank you very much for your work!
I send PRs with you.

**Comment by sktometometo** [comment]
@knorth55 I will work on it,

**Comment by knorth55** [comment]
I rebased `origin/master`

**Comment by knorth55** [comment]
this PR is moved to #1593

**Pull request closed without merging**